src/span-link.ts

```diff
@@ -5,16 +5,12 @@ export function mergeStepFunctionAndLambdaTraces(
   resources: { [key: string]: GeneralResource },
   serverless: Serverless,
 ): void {
-  for (const resourceName in resources) {
-    if (resources.hasOwnProperty(resourceName)) {
```

Comment:
I'm also not sure when this would be false but I think we should check with @kimi-p, he might have some context

Reply:
I do not remember why I put I check there in the first place, but I think that should indeed be true. Thanks for simplifying the logic!